File: app/src/main/java/org/freelesson/pagingsample/RepoSearchResponse.java
package org.freelesson.pagingsample;

import com.google.gson.annotations.SerializedName;

import java.util.Collections;
import java.util.List;

public class RepoSearchResponse {
    @SerializedName("total_count") Integer total = 0;
    @SerializedName("items")
    List<Repo> items= Collections.emptyList();
    Integer nextPage;
}

File: app/src/main/kotlin/org/freelesson/pagingsample/api/RepoSearchResponse.kt
package org.freelesson.pagingsample.api

import org.freelesson.pagingsample.model.Repo
import com.google.gson.annotations.SerializedName

class RepoSearchResponse(
    @SerializedName("total_count") var total: Int = 0,
    @SerializedName("items") var items: List<Repo> = emptyList<Repo>(),
    var nextPage: Int? = null,
)